packages/core/src/components/Sequence/Step.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { FC } from 'react';
+import {
+  Step as MuiStep,
+  StepContent,
+  StepLabel,
+  Typography,
+  makeStyles,
+} from '@material-ui/core';
+import StepperFooter from './StepperFooter';
+
+const useStyles = makeStyles(theme => ({
+  end: {
+    padding: theme.spacing(3),
+  },
+}));
+
+export type StepActions = {
+  showNext?: boolean;
+  canNext?: () => boolean;
+  onNext?: () => void;
+  nextStep?: (current: number, last: number) => number;
+  nextText?: string;
+
+  showBack?: boolean;
+  backText?: string;
+  onBack?: () => void;
+
+  showRestart?: boolean;
+  canRestart?: () => boolean;
+  onRestart?: () => void;
+  restartText?: string;
+};
+
+type StepProps = {
+  title: string;
+  children: React.ReactElement;
+  stepperLength?: number;
+  stepIndex?: number;
+  setStepIndex?: any;
+  stepArray?: number[];
+  setStepArray?: any;
+  end?: boolean;
+  onStepperStepChange?: (prevIndex: number, nextIndex: number) => void;
+  actions?: StepActions;
+};
+
+const Step: FC<StepProps> = ({
+  title,
+  children,
+  end,
+  stepperLength,
+  stepIndex,
+  setStepIndex,
+  stepArray,
+  setStepArray,
```

Comment:
I'm not entirely clear on why the step array exists explicitly and can be set to a new value.

Reply:
so, right now we have steppers that can have a next button that can skip slides depending on answers. so we need to keep a history of the steps for the back button. i should rename this `stepHistory` instead